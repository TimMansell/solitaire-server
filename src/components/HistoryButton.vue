<template>
  <Button
    :is-stacked="isStacked"
    @click="toggleHistory"
    data-test="history-btn"
  >
    History
  </Button>
</template>

<script>
import { mapActions } from 'vuex';
import Button from './Button.vue';

export default {
  name: 'History',
  components: {
    Button,
  },
  props: {
    isStacked: {
      type: Boolean,
      default: false,
    },
  },
  methods: {
    ...mapActions(['toggleHistory']),
  },
};
</script>
