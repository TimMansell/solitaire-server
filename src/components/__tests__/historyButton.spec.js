import { shallowMount } from '@vue/test-utils';
import HistoryButton from '@/components/HistoryButton.vue';

describe('HistoryButton.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(HistoryButton, {
      computed: {
        isEmptyBoard: () => false,
        isOnline: () => true,
        isGamePaused: () => false,
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
