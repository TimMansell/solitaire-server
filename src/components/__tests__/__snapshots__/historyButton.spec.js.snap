// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryButton.vue matches snapshot 1`] = `
<button-stub data-test="history-btn">
  History
</button-stub>
`;
